### PR TITLE
End of round stat update

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Some minor rework to end of round stats.

Changes various stats to be lists organised and recorded by faction, and bullets fired now captures all projectiles (i.e. lasers etc).

Created a new set of stats specifically for Combat Patrol since it's HvH.

## Why It's Good For The Game
Slightly more accurate stats.
Useful stats for HvH.

## Changelog
:cl:
add: Added some HvH specific end of round stats for Combat Patrol, projectiles are counted instead of just bullets
code: tidied up end of round stats
/:cl:

### CODE REVIEW

**Review by BraveMole — COMMENTED** [review]
Is this working? You are not initialising total_projectiles_fired[faction] or total_human_death

**Comment by Lumipharon** [comment]
As far as I can tell, yeah it works fine? Is the way I'm coding it not ideal? 
image

File: code/datums/round_statistics.dm [hunk 1]
```diff
@@ -13,17 +13,16 @@ add [] around this to use it in text.
...
-	var/total_human_deaths = 0
-	var/total_human_revives = 0
+	var/list/total_human_deaths = list()
+	var/list/total_human_revives = list()
 	var/total_human_respawns = 0
 	var/total_xenos_created = 0
-	var/total_humans_created = 0
-	var/total_bullet_hits_on_humans = 0
+	var/list/total_humans_created = list()
+	var/list/total_projectile_hits = list()
```
Comment (TiviPlus) [inline code/datums/round_statistics.dm:25]: should prolly explain why these are lists with an autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/datums/round_statistics.dm (hunk 1 above)
Comment (Lumipharon) [reply code/datums/round_statistics.dm]: updated

**Pull request merged**